src/ir/module-utils.h

```diff
@@ -169,12 +170,18 @@ inline void copyModule(Module& in, Module& out) {
 // to exist.  For example, it is possible to remove a given function and then
 // call this redirect all of its uses.
 template<typename T> inline void renameFunctions(Module& wasm, T& map) {
+  ImportInfo imports(wasm);
+
   // Update the function itself.
   for (auto& pair : map) {
     if (Function* F = wasm.getFunctionOrNull(pair.first)) {
       assert(!wasm.getFunctionOrNull(pair.second));
       F->name = pair.second;
     }
+    // Update any associated GOT.func import.
```

Comment:
Does this belong in `module-utils.h`? There might be a wasm module with `GOT.func` names in it that mean something else, that is, that are not using that dynamic loading mechanism. I think this should be an option so we only do it for for relocatable code, either as a param to this function or another function perhaps?

Reply:
Moved to emscripten-wasm-finalize.